### DISCUSSION TITLE
outputing radial and tangential strain for an "xy" coordinate problem

### DISCUSSION THREAD

**Discussion by afcij** [idaholab/moose#26112] [Q&A Modules: Solid mechanics]
I am doing a 2D tensor mechanics problem on a circular mesh. I have to use Cartesian coordinates (disp_x disp_y), but I need to output radial stress and strain. Using the tensor mechanics master action does not give the correct values for radial stress and strain. I can use the CylindricalRankTwoAux to transform the stress tensor:
```
  (aux variables defined above)
  [./radial_stress]
    type = CylindricalRankTwoAux
    variable = radial_stress
    rank_two_tensor = stress
    index_j = 0
    index_i = 0
    center_point = '0 0 0'
  [../]
  [./hoop_stress] 
    type = CylindricalRankTwoAux
    variable = hoop_stress
    rank_two_tensor = stress
    index_j = 1
    index_i = 1
    center_point = '0 0 0'
  [../]
```
I thought I would be able to do the same for strain. However, I can't find the "rank_two_tensor" name for strain. Can anyone help me with finding this name, or find another method to get radial strain? Here is how I have set up my master action and stress calculation:
```
  [Modules]
    [./TensorMechanics]
      [./Master]
        [./All]
          add_variables = true
          strain = SMALL
          additional_generate_output = 'stress_xx stress_yy strain_xx strain_yy'
          eigenstrain_names = 'thermal_eigenstrain'
          material_output_order = CONSTANT
          material_output_family = MONOMIAL
        [../]
      [../]
    [../]
  []
 [Materials]
  [./stress]
    type = ComputeLinearElasticStress
  [../]
[]
```

**Comment by GiudGiud** [comment]
@gambka
